src/policy/policy.cpp

```diff
@@ -183,16 +183,22 @@ bool IsStandardTx(const CTransaction& tx, const std::optional<unsigned>& max_dat
  *    as potential new upgrade hooks.
  *
  * Note that only the non-witness portion of the transaction is checked here.
+ *
+ * We also check the total number of sigops across the whole transaction.
  */
 bool AreInputsStandard(const CTransaction& tx, const CCoinsViewCache& mapInputs)
 {
     if (tx.IsCoinBase()) {
         return true; // Coinbases don't use vin normally
     }
 
+    unsigned int sigops{0};
     for (unsigned int i = 0; i < tx.vin.size(); i++) {
         const CTxOut& prev = mapInputs.AccessCoin(tx.vin[i].prevout).out;
 
+        sigops += tx.vin[i].scriptSig.GetSigOpCount(true);
+        sigops += prev.scriptPubKey.GetSigOpCount(true);
```

Comment:
// Unlike the block wide sigop limit, the BIP54 per
// transaction limit includes the prevout scriptPubKey.

Reply:
Done, added a sentence to this effect.